server.js

```diff
@@ -26,13 +21,81 @@ const app = express()
 app.use(cors())
 app.use(express.json())
 
+// Our own middleware that checks if the database is connected before going forward to our endpoints
+app.use((req, res, next) => {
+	if (mongoose.connection.readyState === 1) {
+		next();
+	} else {
+		res.status(503).json({ error: "Service unavailable" });
+	}
+});
+
+// Add model for the db
+const NetflixTitle = mongoose.model('NetflixTitle', {
+  show_id: Number,
+  title: String,
+  director: String,
+  cast: String,
+  country: String,
+  date_added: String,
+  release_year: String,
+  rating: String,
+  duration: String,
+  listed_in: String,
+  description: String,
+  type: String
+})
+
+// Seed and save json data to the db
+if (process.env.RESET_DB) {
+
+  const seedDatabase = () => {
+
+    netflixData.forEach(item => {
+      const newNetflixTitle = new NetflixTitle(item)
+      newNetflixTitle.save()
+    })
+  }
+  
+  seedDatabase()
+}
+
 // Start defining your routes here
 app.get('/', (req, res) => {
-  res.send('Hello world')
+  // res.send('Hello from us!')
+  res.send(listEndpoints(app))
+})
+
+// Get all the Netflix movies and shows
+app.get('/netflixtitles', async (req, res) => {
+  const netflixTitles = await NetflixTitle.find(req.query)
+  res.json(netflixTitles)
+})
+
+// Get one Netflix title based on id
+app.get('/netflixtitles/id/:id', async (req, res) => {
+  try {
+    const netflixTitleById = await NetflixTitle.findById(req.params.id)
+    if(!netflixTitleById) {
+      res.status(404).json({
+        response: 'Title not found',
+        success: false,
+      })
+    } else {
+      res.json({
+        response: netflixTitleById,
+        success: true,
+      })
+    }
+  } catch (err) {
+    res.status(400).json({
+      error: 'Id is invalid'
+    })
```

Comment:
I think you can use the fact that you wrote (err) and send it in row 92 as "error: err" and you will get more information from the error message (I haven't tried it out on your code but just realised yesterday how this actually works so could be worth looking at what you get if you send the whole err) :) You can send "message: 'Id is invalid" if you want to as well!